node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/SpareCapacityMaintainer.java

```diff
@@ -68,7 +68,8 @@ public SpareCapacityMaintainer(Deployer deployer,
     @Override
     protected boolean maintain() {
         boolean success = true;
-        if ( ! nodeRepository().zone().getCloud().allowHostSharing()) return success;
+        // Don't need to maintain spare capacity in dynamically provisioned zones; can provision more on demand.
```

Comment:
We'd like to maintain spare capacity in dynamically provisioned zones too - but it will depend on flags.